clippy_lints/src/utils/cargo.rs

```diff
@@ -65,10 +65,10 @@ impl From<json::DecoderError> for Error {
     }
 }
 
-pub fn metadata(manifest_path: Option<String>) -> Result<Metadata, Error> {
+pub fn metadata(manifest_path_arg: &Option<String>) -> Result<Metadata, Error> {
```

Comment:
We don't have a lint that tells us to use Option<&str> instead? We should make one